QRealTime_dialog.py

```diff
@@ -369,44 +381,86 @@ def transformToLayerSRS(self, pPoint):
-        response = requests.get(url,auth=(self.getValue('user'),self.getValue('password')),verify=False)
-        if not response.status_code == 200:
-                return response, table
-        try:
-            data = csv.DictReader(response.text.splitlines(),delimiter=';')
-            out = json.dumps(  [row for row in data]  )
-            tempTable=ast.literal_eval(out)
+    def getTable(self,XFormKey,lastID,topElement,layer,version= 'null'):
+        if lastID:
+            requests.packages.urllib3.disable_warnings()
             table=[]
-            xmlurl="https://kobo.humanitarianresponse.info/assets/"+XFormKey
-            para={'format':'xml'}
-            response2=requests.get(xmlurl,auth=(self.getValue('user'),self.getValue('password')),params=para)
-            ns='{http://www.w3.org/2002/xforms}'
-            xml=response2.content
-            root= ET.fromstring(xml)
+            geoField=''
             imageField=''
-            for bind in root[0][1].findall(ns+'bind'):
-                attrib=bind.attrib
-                fieldName= attrib['nodeset'].split('/')[-1].replace('_', ' ')
-                fieldType=attrib['type']
-                if fieldType[:3]=='bin':
-                    imageField=fieldName
-            submissionIndex=0
-            for submission in tempTable:
-                for key in list(submission):
-                    if key[0]=='_' and key!='_uuid':
-                        submission.pop(key)
-                    if key=='_uuid':
-                        submission['instanceID']=submission[key]
-                        submission.pop(key)
-                    if key==imageField:
-                        imageurl='https://kc.humanitarianresponse.info/attachment/original?media_file='+self.getValue('user')+'/attachments/'+submission[key]
-                        submission[key]=imageurl
-                table.append(submission)
-            print ('table is:',table)
-            return response, table
-        except Exception as e:
-            print ('not able to fetch',e)
-            return response, table
+            url='https://kc.humanitarianresponse.info/api/v1/forms'
+            response = requests.get(url,auth=(self.getValue('user'),self.getValue('password')),verify=False)
+            responseJSON=json.loads(response.text)
+            formID=''
+            for form in responseJSON:
+                if str(form['title'])==XFormKey:
+                    formID=str(form['formid'])
+            urlData='https://kc.humanitarianresponse.info/api/v1/data/'+formID
+            responseData = requests.get(urlData,auth=(self.getValue('user'),self.getValue('password')),verify=False)
+            data=json.loads(responseData.text)
+            uuidList = self.getUUIDList(layer)
+            for submission in data:
+                if submission['meta/instanceID'] not in uuidList:
+                    xmlurl="https://kobo.humanitarianresponse.info/assets/"+submission['_xform_id_string']
```

Comment:
url is not changing in the loop.  i think '_xform_id_string' should be replaced with a variable which iterates through all instance ids.